bundles/AdminBundle/Resources/public/js/pimcore/document/editables/renderlet.js

```diff
@@ -55,9 +55,15 @@ pimcore.document.editables.renderlet = Class.create(pimcore.document.editable, {
             dndManager.addDropTarget(el.getEl(), this.onNodeOver.bind(this), this.onNodeDrop.bind(this));
 
             this.getBody().setStyle({
-                overflow: "auto"
+                overflow: 'auto',
             });
 
+            if (this.config.height !== 'auto') {
+                this.getBodyWrap().setStyle({
```

Comment:
I think we need to move this to `updateDimensions()` and 
```javascript
        this.getBody().setStyle({
            height: "auto"
        });
``` 

needs to be inside the `if` condition.

Reply:
Small update here and some notes:

the default body height has to take into account if there is data set or not, hence the:
```
        if (!this.config.height) {
            this.config.height = this.data.id ? 'auto' : this.defaultHeight;
        }
```
Else the drop zone icon would glitch out and looks broken.

2nd part as for the: 
> I think we need to move this to updateDimensions() and
>        this.getBody().setStyle({
>            height: "auto"
>        });
>needs to be inside the if condition.

As far as I checked thie _getBody().setStyle()_ has to be outside the if, since it needs to be called under any condition.
Else 2 visual bugs appear. If you pass a fixed height, the body needs to draw its height with auto to trigger the parent to make a vertical scrollbar.
If you did not pass a fixed height to the renderlet, the body needs to be a 100% which auto does as well so that the parents will spawn in a vertical dimension to show the complete content of the renderlet body.

WDYT?

Reply:
The `defaultHeight` setting should be only applied if no content was set - that's the only reason for this config. 
I think we should just do it the way the `snippet` editable does, just to be consistent.

Reply:
That's exactly what is being done with: 
```javascript
//line 38++
        if (!this.config.height) {
            this.config.height = this.data.id ? 'auto' : this.defaultHeight;
        }
```

Reply:
> FYI: The snippet is only taking the height: <integer> attribut to render its container if it has no content. As soon as you drop content into a snippet, it will go to height: 'auto'. Is this the desired behavior?

But what's the point then for `defaultHeight`? Actually `defaultHeight` should be the height if no content is set and `height` should be used no matter if content is set or not, right? 
And this is exactly what the `snippet` editable is doing. 😊

Reply:
Sorry, but no. The snippet is not doing what you think it should.
If you set
```twig
{{ pimcore_snippet('my-snippet', {height: 500}) }}
```

Then you will have a snippet d'n'd area being 500px height initially (or 100px as fallback if you don't pass the height attribute). But as soon as you drag a snippet onto that, then the 500px from this example will be replaced by height => auto and the snippet is as heigh as the content needs it to be.

IMHO: 
* defaultHeight => fallback if no explicit height has been set via call (e.g. pimcore_renderlet('my-renderlet, {height: 500})) => content is heigher than defaultHeight => height is set to auto to fully show the content of the element without scrollbars
* height => height of the container (snippet or renderlet) set to exactly this value, wether content is set or not (if set, and content is heigher => show scrollbars)

:D do you agree on that or are we still miles apart?